lib/Minz/Request.php

```diff
@@ -294,8 +348,41 @@ private static function magicQuotesOff() {
 		}
 	}
 
+	/**
+	 * @return array
+	 */
+	private static function getHeaders() {
+		$headers = array();
+		foreach ($_SERVER as $key => $value) {
+			$headers[$key] = $value;
+		}
+		return $headers;
+	}
+
+	/**
+	 * @return mixed
+	 */
+	public static function getHeader($header, $default = null) {
+		if (array_key_exists($header, static::$headers)) {
+			return static::$headers[$header];
+		}
+		return $default;
+	}
+
+	/**
+	 * @return boolean
+	 */
 	public static function isPost() {
-		return isset($_SERVER['REQUEST_METHOD']) &&
-			$_SERVER['REQUEST_METHOD'] === 'POST';
+		return 'POST' === static::getHeader('REQUEST_METHOD');
+	}
+
+	/**
+	 * @return array
+	 */
+	public static function getPreferredLanguage() {
```

Comment:
I'm not sure why you added this method (it's not a problem)

Reply:
This method extract the preferred language from the header. You could have something like `fr-CH, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5` in the header and you'll end up with an array containing `fr-CH`, `fr`, `en`, `de` and `*`. When those values are retrieved, we can match the first language that we have a translation for. In this case, `fr`.

Reply:
I understand the code, what I mean is that you don't use it anywhere so I don't know why you added it :wink:

Reply:
I don't use now because my goal was to introduce some helper methods. This one is to allow FRSS to switch language automatically to the one preferred by the user. It will become useful on the public connection interface. When the user is not connected, FRSS does not know the user config. Thus relying on its preferred language is the sole way to change the interface.

Reply:
The code to do such thing must be added though :)